solnlib/bulletin_rest_client.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from solnlib import splunk_rest_client as rest_client
+from splunklib import binding
+from typing import Optional
+import json
+
+
+__all__ = ["BulletinRestClient"]
+
+
+class BulletinRestClient:
+    READ_ENDPOINT = "/services/admin/messages"
+    CREATE_DELETE_ENDPOINT = "/services/messages"
+
+    headers = [("Content-Type", "application/json")]
+
+    class LogLevel:
+        INFO = "info"
+        WARNING = "warn"
+        ERROR = "error"
+
+    def __init__(
+        self,
+        message_name: str,
+        session_key: str,
+        logger: Optional[logging.Logger] = None,
+        **context: dict,
+    ):
+
+        self.message_name = message_name
+        self.session_key = session_key
+
+        if logger:
+            self.logger = logger
+        else:
+            self.logger = logging
```

Comment:
If the type we're expecting is `logging.Logger`, shouldn't this variable have the value as `logging.Logger`? 
Moreover, we could get rid of the redundant if...else condition if we pass the default value in method declaration.

Reply:
Thanks, I've missed that. Logger is no longer used so I removed it